### PR TITLE
implement DriverDaoJdbcImpl and CarDaoJdbcImpl

### PR DESCRIPTION
**Pull request by IwalkInSilence**

### CODE REVIEW

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create car "
+                    + car.toString() + " in DB", e);
```
Comment (Artificial-Friend) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: .toString() will be called automatically

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT * "
```
Comment (Artificial-Friend) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: Avoid star select since columns may be added and will be redundant in this query

Comment (Mandybur800) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: try to avoid wildcard `*`

**Review by Artificial-Friend — APPROVED** [review]

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car with ID " + id, e);
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        Car car = null;
```
Comment (Mandybur800) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: remove this

**Comment by Artificial-Friend** [comment]
init_db must be init_db.sql

**Review by Artificial-Friend — APPROVED** [review]

**Review by Mandybur800 — CHANGES_REQUESTED** [review]

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+                + "ON m.id = c.manufacturer_id "
+                + "WHERE c.id = ? AND c.deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+                List<Driver> drivers = getDrivers(id);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: you are calling getDrivers() twice

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted = false AND m.deleted = false ";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> carList = new ArrayList<>();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                carList.add(get(id).get());
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java]: let's build our car here and use getDrivers() for each of them

File: src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+                + "INNER JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "INNER JOIN drivers d ON d.id = cd.driver_id "
+                + "WHERE cd.driver_id = ? AND d.deleted = FALSE AND c.deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/dao/jdbc/CarDaoJdbcImpl.java:138]: let's user getDrivers() all other information to build each car you have

**Review by dmytro-hryhoruk — CHANGES_REQUESTED** [review]

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**